pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -167,6 +197,54 @@ func (c *podIpMetricsCollector) listIpAddresses(dir string) error {
 	return nil
 }
 
+func getIPv4(s string) (bool, string) {
+	names := strings.Split(s, "/")
+	f := names[len(names)-1]
+	ip := net.ParseIP(f)
+	if ip == nil || ip.To4() == nil {
+		return false, ""
+	}
+	return true, f
+}
+
+// After the ip is removed, it will be put into the reuseMap.
+// After the ip is reused, the reuseMap will update the ip reuse time.
+// The reuseMap will not record any ip that is being used, only keeps the reused one.
+func (c *podIpMetricsCollector) updateReuseIps(e fsnotify.Event, reuseMap map[string]*ipReuse, f string, bucketKeys []float64) {
+	reuseIp, ok := reuseMap[f]
+	if e.Op&fsnotify.Remove == fsnotify.Remove {
+		if !ok {
+			reuseMap[f] = &ipReuse{time.Now(), 0}
+		} else {
+			reuseIp.podKilledTimestamp = time.Now()
+		}
+	} else if e.Op&fsnotify.Create == fsnotify.Create {
+		if ok {
+			oldT := reuseIp.podKilledTimestamp
+			diff := uint64(time.Now().Sub(oldT).Milliseconds())
+			reuseIp.ipreuseTime = float64(diff)
+
+			c.reuseIps.count += 1
+			c.reuseIps.sum += float64(diff)
+			if diff < c.reuseIps.min {
+				c.reuseIps.min = diff
+			}
+			c.fillBuckets(diff, bucketKeys)
+		}
+	}
+}
+
+func (c *podIpMetricsCollector) fillBuckets(diff uint64, bucketKeys []float64) {
+	prev := float64(0)
+	fill := false
+	for _, bound := range bucketKeys {
+		if fill || (prev <= float64(diff) && float64(diff) < bound) {
```

Comment:
You aren't really using prev. This line can simply be:
if float64(diff) < bound {}
You don't even need 'fill'.

Reply:
Done.